### ISSUE TITLE
openrc-0.40.2 supervise-deamon healthcheck_timer file descriptor leak, plus infinite supervise-deamon --stop kill 0 loop

### ISSUE THREAD

**Issue by zmedico** [OpenRC/openrc#271]
The supervise-daemon leaks one `'/run/openrc/exclusive/${RC_SVCNAME} (deleted)'` file descriptor for each healthcheck_timer period.

After the file descriptors are exhausted, an attempt to stop the service triggers an infinite kill 0 loop in the `start-stop-daemon ${RC_SVCNAME} --stop` call with `retry="SIGTERM/2/SIGTERM/3/SIGKILL/5"`:
```
# strace -p 7955
strace: Process 7955 attached
nanosleep({tv_sec=0, tv_nsec=1}, NULL)  = 0
kill(22293, SIG_0)                      = 0
nanosleep({tv_sec=0, tv_nsec=1}, NULL)  = 0
kill(22293, SIG_0)                      = 0
```
Meanwhile, the start-stop-daemon instance with exhausted file descriptors is in a blocking wait call:
```
# strace -p 22293
strace: Process 22293 attached
wait4(-1,
```

**Comment by williamh** [comment]
The infinite loop you are seeing in the second instance is because it is waiting for the first one to go down. The first one is stuck in the wait call due to exhausting the file descriptors. I am running a long-running test on master with the patch above applied to make sure that the file descriptors do not leak any longer. If they do not, this should also fix the shutdown. I will verify and update in the morning.

**Comment by williamh** [comment]
This was fixed in 0.40.3.
